raster-foundry/raster-foundry
# Add analyses to v2 project publish page

## Overview
Added analyses to the new project publish page and fixed a bunch of bugs with the url generation and copying.
Fix /tool-runs projectLayerId parameter

### Checklist

- ~Styleguide updated, if necessary~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo
![image](https://user-images.githubusercontent.com/4392704/56811847-aabbb980-6807-11e9-8250-774b547395a0.png)

## Testing Instructions
- Re-assemble your api jar in SBT
- Create a project
- Create three layers on the project
- Create a template which can be used with project layers
- Create 6 analyses on one layer, and 3 analyses on a second
- Verify that when you go to the publish page, each layer lists the number of linked analyses
- Select the layers by clicking anywhere on the list item
- Verify that the urls in the main view make sense, both in the text box and when clicking the copy button and pasting elsewhere
- Make sure the project privacy settings are private
- Verify that the map token is populated and used in all the urls (published page and tile layer urls)
- Verify that the share page works
- Verify that verify that the tile urls work
- Change the project privacy settings to public
- Verify that the map token is no longer in any of the urls or copied text from buttons
- Verify that the share page works
- Verify that the tile urls work

Closes #4860
